homeassistant/components/uvc/camera.py

```diff
@@ -1,103 +1,90 @@
 """Support for Ubiquiti's UVC cameras."""
 import logging
-import socket
+from typing import Optional
 
 import requests
-from uvcclient import camera as uvc_camera, nvr
 import voluptuous as vol
 
 from homeassistant.components.camera import PLATFORM_SCHEMA, SUPPORT_STREAM, Camera
-from homeassistant.const import CONF_PORT, CONF_SSL
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
 from homeassistant.exceptions import PlatformNotReady
 import homeassistant.helpers.config_validation as cv
 
-_LOGGER = logging.getLogger(__name__)
-
-CONF_NVR = "nvr"
-CONF_KEY = "key"
-CONF_PASSWORD = "password"
+from . import CONF_KEY, CONF_NVR, CONF_PORT, CONF_SSL, DEFAULT_PORT, DEFAULT_SSL
 
-DEFAULT_PASSWORD = "ubnt"
-DEFAULT_PORT = 7080
-DEFAULT_SSL = False
+_LOGGER = logging.getLogger(__name__)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_NVR): cv.string,
         vol.Required(CONF_KEY): cv.string,
-        vol.Optional(CONF_PASSWORD, default=DEFAULT_PASSWORD): cv.string,
         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
         vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
     }
 )
 
 
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+):
+    """Set up camera entry."""
+    return setup_platform(hass, entry.data, async_add_entities)
+
+
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Discover cameras on a Unifi NVR."""
-    addr = config[CONF_NVR]
-    key = config[CONF_KEY]
-    password = config[CONF_PASSWORD]
-    port = config[CONF_PORT]
-    ssl = config[CONF_SSL]
-
     try:
-        # Exceptions may be raised in all method calls to the nvr library.
-        nvrconn = nvr.UVCRemote(addr, port, key, ssl=ssl)
-        cameras = nvrconn.index()
-
-        identifier = "id" if nvrconn.server_version >= (3, 2, 0) else "uuid"
-        # Filter out airCam models, which are not supported in the latest
-        # version of UnifiVideo and which are EOL by Ubiquiti
         cameras = [
             camera
-            for camera in cameras
-            if "airCam" not in nvrconn.get_camera(camera[identifier])["model"]
+            for camera in requests.get(
```

Comment:
We don't allow device specific interface code or data parsing in home assistant. That needs to be moved to a standalone library published on pypi.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices